Automattic/wp-calypso
# Site Profiler: Redirect to final report

Do not call basic endpoint when having the hash



Resolves https://github.com/Automattic/dotcom-forge/issues/7116

Depend on D151203-code which is already in trunk ✅ 

## Proposed Changes

* Automatically redirect to the URL with the hash and the `final_url` params to display the advanced report
* Do not call the basic endpoints if the hash is present, so we are in the advanced report
* Only call the advanced endpoint if the hash is present, so it is always valid
* Display the Results Header if we have advanced data, do not depend on the basic metrics to be present 
* Do not show the landing page when we are displaying the advanced report

## Why are these changes being made?


* To fix issues with the UX of the tool

## Testing Instructions



* Apply this patch
* Make sure the linked diff D151203-code is applied and you are sandboxing your environment - Already in trunk
* Navigate to the Site Profiler and check a site
* Make sure that after populating the site and clicking on submit, the browser is redirected to a URL with the `hash` 
* Make sure the advanced report is displayed as expected
* Refresh the advanced report
* Make sure the landing page is not displayed anymore and, instead, the Loading screen is displayed for a split of a second
* Populate the get report form and submit it. Wait for the email to be received
* Navigate to the report sent to the email. Copy the URL except the domain, and replace it with your local testing environment.
* Check the advanced report is displayed and the landing page is not displayed.

## Pre-merge Checklist
